include/logger/formatter.hpp

```diff
@@ -68,5 +104,37 @@ std::string logger::IFormatter::FormatData(std::string msg,
 {
   return this->format(msg, level, logger_name, log_credentials);
 }
-};// namespace logger
+
+// Definitions of class DefaultFormatter.
+std::string logger::DefaultFormatter::square_bracket_style_format_msg(std::string msg,
+  logger::LEVEL level,
+  std::string &logger_name,
+  std::shared_ptr<logger::ExtractedLogCredentials> &log_credentials)
+{
+  std::string formatted_string;
+  std::vector<std::string> message_chunk;
```

Comment:
This suggestion is not necessary to implement as looking at the size of vector i'ts very trivial.
just want to tell you the background of it.
vector has a array based implementation, so internally it needs to have some size the default is 0.
every time .push_back() method called it check for size,  and if its not sufficient to hold new element.
then its reallocate memory with + 1 in size , then copies existing element into newly allocated array then at the end adds the new element. 
to avoid this you can use 
.reserve(some value) method.
looking at the cod you have max 8 element-> this things should be used when you know the number of element in advance.

also you are not doing any random access of vector.  so std::list will be the nice option to use instead of vector.

sorry if the information is too much, but i guess this stuff are necessary to know